cloudbuild.yaml

```diff
@@ -6,45 +6,42 @@ options:
 steps:
   - name: 'gcr.io/cloud-builders/bazel'
     env:
-    - IMAGE_REGISTRY=gcr.io
-    - IMAGE_REPO=k8s-staging-cloud-provider-gcp
-    - IMAGE_TAG=${_PULL_BASE_REF}
+      - IMAGE_REGISTRY=gcr.io
+      - IMAGE_REPO=k8s-staging-cloud-provider-gcp
+      - IMAGE_TAG=${_PULL_BASE_REF}
     args:
       - run
       - //cmd/cloud-controller-manager:publish
   - name: 'gcr.io/cloud-builders/bazel'
     env:
-    - IMAGE_REGISTRY=gcr.io
-    - IMAGE_REPO=k8s-staging-cloud-provider-gcp
-    - IMAGE_TAG=${_PULL_BASE_REF}
+      - IMAGE_REGISTRY=gcr.io
+      - IMAGE_REPO=k8s-staging-cloud-provider-gcp
+      - IMAGE_TAG=${_PULL_BASE_REF}
     args:
       - run
       - //cmd/gcp-controller-manager:publish
+  # build gke-exec-auth-plugin binary
   - name: 'gcr.io/cloud-builders/bazel'
-    env:
-    - IMAGE_REGISTRY=gcr.io
-    - IMAGE_REPO=k8s-staging-cloud-provider-gcp
-    - IMAGE_TAG=${_PULL_BASE_REF}
     args:
-      - run
-      - //cmd/gke-exec-auth-plugin:publish
+      - build
+      - //cmd/gke-exec-auth-plugin
+  # build gke-gcloud-auth-plugin binary
   - name: 'gcr.io/cloud-builders/bazel'
-    env:
-    - IMAGE_REGISTRY=gcr.io
-    - IMAGE_REPO=k8s-staging-cloud-provider-gcp
-    - IMAGE_TAG=${_PULL_BASE_REF}
     args:
-      - run
-      - //cmd/gke-gcloud-auth-plugin:publish
+      - build
+      - //cmd/gke-gcloud-auth-plugin
   # build auth-provider-gcp binary
   - name: 'gcr.io/cloud-builders/bazel'
     args:
       - build
       - //cmd/auth-provider-gcp
-# upload auth-provider-gcp binary
+# upload auth-provider-gcp. gke-gcloud-auth-plugin and gke-exec-auth-plugin binaries
 artifacts:
   objects:
-    location: 'gs://k8s-staging-cloud-provider-gcp/auth-provider-gcp-linux-amd64#$_PULL_BASE_REF'
-    paths: ['bazel-bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp']
+    location: 'gs://k8s-staging-cloud-provider-gcp/auth-linux-amd64-$_PULL_BASE_REF'
+    paths:
+      - 'bazel-bin/cmd/auth-provider-gcp/auth-provider-gcp_/auth-provider-gcp'
+      - 'bazel-bin/cmd/gke-gcloud-auth-plugin/gke-gcloud-auth-plugin_/gke-gcloud-auth-plugin'
+      - 'bazel-bin/cmd/gke-exec-auth-plugin/gke-exec-auth-plugin_/gke-exec-auth-plugin'
```

Comment:
These aren't working because `bazel-bin` is actually a symlink into the bazel cache which isn't being persisted under the workspace volume.

We either need to:
1. copy them out to a path like `bin/<binary>` within the step container
2. ensure the bazel cache is persisted (and hopefully the symlinks can be followed ....)
3. run gsutil cp ourselves within the step container

I'd recommend 3) or 1). 3) probably makes the most sense but I'm not sure if the bazel image has gsutil, it should have cp. Spoke to @ndixita about this yesterday.